### PR TITLE
[Docs] Chatgpt Agent Setup

### PR DESCRIPTION
**Pull request by Future-Outlier**
## Docs link
https://flyte--4992.org.readthedocs.build/en/4992/deployment/agents/index.html

## Tracking Issues
https://github.com/flyteorg/flyte/issues/3936

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] All commits are signed-off.

## Related PRs
https://github.com/flyteorg/flytesnacks/pull/1539

### CODE REVIEW

**Comment by Future-Outlier** [comment]
I fail to build docs in my local environments too.
image

File: docs/deployment/agents/chatgpt.rst [hunk 1]
```diff
@@ -0,0 +1,140 @@
+.. _deployment-agent-setup-chatgpt:
+
+Chatgpt agent
```
Comment (neverett) [inline docs/deployment/agents/chatgpt.rst:3]:
```suggestion
ChatGPT agent
```

File: docs/deployment/agents/chatgpt.rst [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+=================
+
+This guide provides an overview of how to set up the Chatgpt agent in your Flyte deployment.
```
Comment (neverett) [inline docs/deployment/agents/chatgpt.rst:6]:
```suggestion
This guide provides an overview of how to set up the ChatGPT agent in your Flyte deployment.
```

File: docs/deployment/agents/chatgpt.rst [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+Please note that you have to setup the openai api key in the agent server to to run the Chatgpt tasks.
```
Comment (neverett) [inline docs/deployment/agents/chatgpt.rst:7]:
```suggestion
Please note that you have to set up the OpenAI API key in the agent server to to run ChatGPT tasks.
```

File: docs/deployment/agents/chatgpt.rst [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+              - chatgpt: agent-service
+
+        plugins:
+          agent-service:
+            supportedTaskTypes:
+            - chatgpt
+
+    .. group-tab:: Flyte core
+
+      Create a file named ``values-override.yaml`` and add the following configuration to it.
```
Comment (neverett) [inline docs/deployment/agents/chatgpt.rst:44]:
```suggestion
      Create a file named ``values-override.yaml`` and add the following configuration to it:
```

File: docs/deployment/agents/chatgpt.rst [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                  container: container
+                  sidecar: sidecar
+                  container_array: k8s-array
+                  chatgpt: agent-service
+            plugins:
+              agent-service:
+                supportedTaskTypes:
+                - chatgpt
+
+Add the OPENAI API token
```
Comment (neverett) [inline docs/deployment/agents/chatgpt.rst:70]:
```suggestion
Add the OpenAI API token
```

File: docs/deployment/agents/chatgpt.rst [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+-------------------------------
+
+You have to set the Databricks token to the Flyte configuration.
```
Comment (neverett) [inline docs/deployment/agents/chatgpt.rst:73]: Should this be removed?

File: docs/deployment/agents/chatgpt.rst [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+1. Install flyteagent pod using helm
```
Comment (neverett) [inline docs/deployment/agents/chatgpt.rst:75]:
```suggestion
1. Install flyteagent pod using helm:
```

File: docs/deployment/agents/chatgpt.rst [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+  
+.. code-block::
+  
+  helm repo add flyteorg https://flyteorg.github.io/flyte
+  helm install flyteagent flyteorg/flyteagent --namespace flyte
+
+2. Get the base64 value of your OPENAI API token.
```
Comment (neverett) [inline docs/deployment/agents/chatgpt.rst:82]:
```suggestion
2. Get the base64 value of your OPENAI API token:
```

File: docs/deployment/agents/chatgpt.rst [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+
+.. code-block::
+
+  echo -n "<OPENAI_API_TOKEN>" | base64
+
+3. Edit the flyteagent secret
```
Comment (neverett) [inline docs/deployment/agents/chatgpt.rst:88]:
```suggestion
3. Edit the flyteagent secret:
```

File: docs/deployment/agents/chatgpt.rst [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+
+    .. code-block:: bash
+
+      helm upgrade <RELEASE_NAME> flyte/flyte-core -n <YOUR_NAMESPACE> --values values-override.yaml
+
+    Replace ``<RELEASE_NAME>`` with the name of your release (e.g., ``flyte``)
+
+    and ``<YOUR_NAMESPACE>`` with the name of your namespace (e.g., ``flyte``).
+
+For Chatgpt agent on the Flyte cluster, see `Chatgpt agent <https://docs.flyte.org/en/latest/flytesnacks/examples/chatgpt_agent/index.html>`_.
```
Comment (neverett) [inline docs/deployment/agents/chatgpt.rst:140]:
```suggestion
For ChatGPT agent on the Flyte cluster, see `ChatGPT agent <https://docs.flyte.org/en/latest/flytesnacks/examples/chatgpt_agent/index.html>`_.
```

File: docs/deployment/agents/index.md [hunk 11]
```diff
@@ -17,6 +17,8 @@ If you are using a managed deployment of Flyte, you will need to contact your de
 
 * - {ref}`Airflow Agent <deployment-agent-setup-airflow>`
   - Configuring your Flyte deployment for the Airflow agent
+* - {ref}`ChatGPT Agent <deployment-agent-setup-databricks>`
```
Comment (neverett) [inline docs/deployment/agents/index.md:20]:
```suggestion
* - {ref}`ChatGPT Agent <deployment-agent-setup-chatgpt>`
```

**Review by neverett — CHANGES_REQUESTED** [review]
A few changes requested, otherwise looks good, thanks for adding this!

File: docs/deployment/agents/chatgpt.rst (hunk 6 above)
Comment (Future-Outlier) [reply docs/deployment/agents/chatgpt.rst:73]: yes, thank you

**Review by neverett — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]

**Pull request merged**